oauth2_provider/oauth2_backends.py

```diff
@@ -148,6 +151,16 @@ def create_authorization_response(self, request, scopes, credentials, allow):
         except oauth2.OAuth2Error as error:
             raise OAuthToolkitError(error=error, redirect_uri=credentials["redirect_uri"])
 
+    def create_device_authorization_response(self, request: HttpRequest):
```

Comment:
Are you sure this is a django.http.HttpRequest and not an oauthlib.common.Request?